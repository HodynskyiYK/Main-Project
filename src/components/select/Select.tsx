import React, { FC } from 'react'
import styles from './Select.module.scss'
import { ISelect } from './selectTypes'

export const Select: FC<ISelect> = (
    {
        selectName,
        labelText
    }) => {

    return (
        <>
            {
                labelText && (
                    <label
                        className={styles.Label}
                        htmlFor={selectName}
                    >{labelText}</label>
                )
            }
            <select
                className={styles.Select}
                name={selectName}
                id={selectName}
            >
                <option value="SelectGenre">Select Genre</option>
            </select>
        </>
    )
}
