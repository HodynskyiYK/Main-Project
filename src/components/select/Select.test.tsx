import React from 'react'
import { Select } from './Select'
import { render, screen } from '@testing-library/react'

describe('Select tests', () => {
    let options: string[]
    beforeEach(() => {
        options = ['option1', 'option2']
    })

    test('To match snapshot', () => {
        const {asFragment} = render(<Select selectName={'select'} options={options} />)
        expect(asFragment()).toMatchSnapshot()
    })

    test('label text', () => {
        render(<Select selectName={'select'} options={options} labelText={'label text'} />)
        const label = screen.getByText(/label text/i)
        expect(label).toBeInTheDocument()
    })
})