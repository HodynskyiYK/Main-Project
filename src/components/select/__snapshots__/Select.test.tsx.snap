// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select tests To match snapshot 1`] = `
<DocumentFragment>
  <select
    class="Select"
    id="select"
    name="select"
  >
    <option
      value="option1"
    >
      option1
    </option>
    <option
      value="option2"
    >
      option2
    </option>
  </select>
</DocumentFragment>
`;
